arangod/Aql/Condition.cpp

```diff
@@ -1754,8 +1754,13 @@ bool Condition::canRemove(ExecutionPlan const* plan, ConditionPart const& me,
         }
       }
     }
-  } catch (...) {
-    // simply ignore any errors and return false
+  } catch (std::exception const& ex) {
```

Comment:
What about callers this function, do they except exceptions from this code?

Reply:
Previously all exceptions were caught and silenced, without letting the caller know about it.
Now we do effectively the same (catching all exceptions of type `std::exception`, which is the only type of exception that could be thrown by any of the code in question). We are still not returning any indication of the exception to the caller, but simply log it in trace mode.
It is normally not a problem if an exception is caught here and silenced. It only means that we will miss some potential optimization opportunity for an AQL condition expression. But the query should still function normally with or without the exception happening.